Source/Xamarin/Prism.Ninject.Forms/Extensions/DependencyServiceProvider.cs

```diff
@@ -0,0 +1,43 @@
+﻿using Ninject.Activation;
+using System;
+using System.Reflection;
+using Xamarin.Forms;
+
+namespace Prism.Ninject.Extensions
+{
+    /// <summary>
+    /// Creates instances of services using the
+    /// <see cref="Xamarin.Forms.DependencyService"/>
+    /// </summary>
+    public class DependencyServiceProvider : IProvider
+    {
+        /// <summary>
+        /// Constructs a <see cref="DependencyServiceProvider"/>
+        /// </summary>
+        /// <param name="type">The type that this provider resolves</param>
+        public DependencyServiceProvider(Type type)
+        {
+            Type = type;
+        }
+
+        /// <inheritDoc />
+        public Type Type { get; private set; }
+
+        /// <inheritDoc />
+        public object Create(IContext context)
+        {
+            return ResolveFromDependencyService(Type);
+        }
+
+        private static object ResolveFromDependencyService(Type targetType)
```

Comment:
Does the actual object creation from DependencyService. Code is cribbed from the Unity version.